entity-view/api/src/main/java/com/blazebit/persistence/view/With.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.view;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import com.blazebit.persistence.view.With.RepeatableWith;
+
+/**
+ * This annotation invokes the {@link CteProvider} to bind the {@code cte} class
+ * to the query.
+ *
+ * @author Giovanni Lovato
+ * @since 1.4.0
+ */
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(RepeatableWith.class)
+public @interface With {
+
+    /**
+     * The CTE class.
+     *
+     * @return The CTE class
+     */
+    Class<?> cte();
+
+    /**
+     * A flag set to {@code true} for recursive CTEs.
+     *
+     * @return true if recursive, false otherwise
+     */
+    boolean recursive() default false;
```

Comment:
I don't think this is necessary, the provider will call with or withRecursive

Reply:
The idea was that the `With` annotation would already call `with` and provide a `FullSelectCTECriteriaBuilder` instead of a `CTEBuilder`. If we provide a `CTEBuilder` instead, then yeah there's no need for `cte` and `recursive` attributes!